kmodes/kprototypes.py

```diff
@@ -513,3 +527,16 @@ def _split_num_cat(X, categorical):
                                if ii not in categorical]]).astype(np.float64)
     Xcat = np.asanyarray(X[:, categorical])
     return Xnum, Xcat
+
+
+def _validate_sample_weights(sample_weights, n_samples):
```

Comment:
If we're enabling this for both KModes and KPrototypes, I suggest moving this method to the former.

Reply:
Noted (for a possible follow-up PR)!